driver/registry_default_test.go

```diff
@@ -295,12 +295,91 @@ func TestDriverDefault_Hooks(t *testing.T) {
 	})
 
 	t.Run("type=settings", func(t *testing.T) {
-		conf, reg := internal.NewFastRegistryWithMocks(t)
-		conf.MustSet(config.ViperKeySelfServiceVerificationEnabled, true)
+		// AFTER hooks
+		for _, tc := range []struct {
+			uc     string
+			prep   func(conf *config.Config)
+			expect func(reg *driver.RegistryDefault) []settings.PostHookPostPersistExecutor
+		}{
+			{
+				uc:     "No hooks configured",
+				prep:   func(conf *config.Config) {},
+				expect: func(reg *driver.RegistryDefault) []settings.PostHookPostPersistExecutor { return nil },
+			},
+			{
+				uc: "Only verify hook configured for the strategy",
+				prep: func(conf *config.Config) {
+					conf.MustSet(config.ViperKeySelfServiceVerificationEnabled, true)
+					// I think this is a bug as there is a hook named verify defined for both profile and password
+					// strategies. Instead of using it, the code makes use of the property used above and which
+					// is defined in an entirely different flow (verification).
+				},
```

Comment:
@aeneasr: Is it really the desired this way?

Reply:
Not sure I understand, could you elaborate?

Reply:
I mean that there are hooks defined for the same purpose.

My understanding of the schema and the documentation is that one would have to configure a verification hook, like this:

```.yaml
settings:
  ...
  after:
    password:
      hooks:
        - hook: verify
```

But the code does not take this into account at all. Instead it checks, whether the `enabled` property of the `verification` flow: 

```.go
func (m *RegistryDefault) PostSettingsPostPersistHooks(ctx context.Context, settingsType string) (b []settings.PostHookPostPersistExecutor) {
	initialHookCount := 0
	if m.Config(ctx).SelfServiceFlowVerificationEnabled() {   <-------------------------------
		b = append(b, m.HookVerifier())
		initialHookCount = 1
	}
...
```

Reply:
Btw, some other flow implementations do the same

Reply:
Ah I see - the reasoning here is that we enable the verification hook per default when the flow is enabled as it would not really make sense otherwise to be enabled. I get however that this is a bit confusing here. Basically, the idea is: why would you NOT want to send an email when you have verification enabled?

I am however open to change this or at least remove the "verify" hook from the available config options.

Reply:
Looking at the current docs I think it would make sense to remove the "verify" hook from the config schema (in a a separate PR) -> https://github.com/ory/kratos/issues/1302

Reply:
> Basically, the idea is: why would you NOT want to send an email when you have verification enabled?

Because we now have web hooks ;) and one would maybe like to use other mechanisms? ¯\_(ツ)_/¯